File: Leetcode-Submissions/0069-sqrtx/solution.cpp
int sqrt(int x){
    int s = 0;
    int e = x;
    long long int mid = s + (e - s)/2;
    
    int ans = -1;
    while(s <= e){
        long long int sqr = mid * mid;
   
        if(sqr == x){
         return mid;
    }
       else if ( sqr < x){
           ans  = mid;
           s = mid + 1;
       }
        else{
            e = mid - 1;
        }
        mid = s + (e - s)/2;
    }
    return ans;
    
}

class Solution {
public:int mySqrt(int x) {
             return sqrt(x);
    
  }  
};

File: Leetcode-Submissions/0069-sqrtx/solution.java
class Solution {
    public int mySqrt(int x) {
        int a = 1, b = x, mid = a + (x - a) / 2;
        int sqrt = 0;
        while(a <= b){
            if(mid == x / mid){
                return mid;
            }
            else if(mid < x / mid){
                sqrt = mid;
                a = mid + 1;
            }
            else{
                b = mid - 1;
            }
            mid = a + (b - a) / 2;

        }
        return sqrt;

    }
}
